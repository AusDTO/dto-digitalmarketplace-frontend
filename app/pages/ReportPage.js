import React, { Component } from 'react'
import { connect } from 'react-redux'
import { withRouter, Switch, Route, Redirect } from 'react-router-dom'
import PropTypes from 'prop-types'
import ReportView from '../components/Reports/ReportView'
const itemData = require('../components/Reports/reportdata.json')
import { rootPath } from '../routes'

class ReportPage extends Component {
  constructor(props) {
    super(props)
    this.state = {
      dataLoaded: false
    }
  }

  render() {
    const { match } = this.props

    return (
      <div id="report-page">
        <Switch>
          <Route
            exact
            path={match.url}
            render={() => <ReportView {...this.props} data={itemData} dataLoaded={this.state.dataLoaded} />}
          />
          <Redirect to={{ pathname: `${rootPath}/reports` }} />
        </Switch>
      </div>
    )
  }
}

ReportPage.propTypes = {
  data: PropTypes.shape({
    title: PropTypes.string,
    items: PropTypes.arrayOf(
      PropTypes.shape({
        text: PropTypes.oneOfType([PropTypes.string, PropTypes.array]),
        image: PropTypes.string
      })
    )
  }),
  match: PropTypes.object
}

ReportPage.defaultProps = {
  data: {
    title: 'Marketplace Reports',
    items: [
      {
        text: '',
        image: ''
      }
    ]
  },
  match: () => ({ url: `${rootPath}/reports` })
}

const matchStateToProps = state => ({
  media: state.media
})

export default withRouter(connect(matchStateToProps)(ReportPage))