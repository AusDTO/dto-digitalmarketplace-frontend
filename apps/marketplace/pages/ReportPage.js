import React, { Component } from 'react'
import { connect } from 'react-redux'
import { withRouter, Switch, Route, Redirect } from 'react-router-dom'
import PropTypes from 'prop-types'
import ReportView from '../components/Reports/ReportView'
import { rootPath } from '../routes'

const itemData = require('../components/Reports/reportdata.json')

export class ReportPageContainer extends Component {
  constructor(props) {
    super(props)
    this.state = {
      dataLoaded: false
    }
  }

  render() {
    const { match } = this.props

    return (
      <div id="report-page">
        <Switch>
          <Route
            exact
            path={match.url}
            render={() => <ReportView {...this.props} data={itemData} dataLoaded={this.state.dataLoaded} />}
          />
          <Redirect to={{ pathname: `${rootPath}/reports` }} />
        </Switch>
      </div>
    )
  }
}

ReportPageContainer.propTypes = {
  data: PropTypes.shape({
    title: PropTypes.string,
    date: PropTypes.string,
    items: PropTypes.arrayOf(
      PropTypes.shape({
        text: PropTypes.oneOfType([PropTypes.string, PropTypes.array]),
        image: PropTypes.string
      })
    )
  }),
  match: PropTypes.object
}

ReportPageContainer.defaultProps = {
  data: {
    title: 'Marketplace Reports',
    date: null,
    items: [
      {
        heading: '',
        media: '',
        subitems: []
      }
    ]
  },
  match: () => ({ url: `${rootPath}/reports` })
}

const matchStateToProps = state => ({
  media: state.media
})

export default withRouter(connect(matchStateToProps)(ReportPageContainer))
