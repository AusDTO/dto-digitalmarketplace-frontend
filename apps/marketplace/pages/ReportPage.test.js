import React from 'react'
import ReactShallowRenderer from 'react-test-renderer/shallow'
import { Provider } from 'react-redux'
import expect from 'expect'
import configureStore from '../store'
import { ReportPageContainer } from './ReportPage'

jest.mock('shared/Icon/_getIcons')

describe('Test suite for ReportPageContainer component', () => {
  const state = {}

  const store = configureStore()
  test('ReportPageContainer renders without errors', () => {
    const renderer = new ReactShallowRenderer()
    const tree = renderer.render(
      <Provider store={store}>
        <ReportPageContainer {...state} />
      </Provider>
    )

    expect(tree).toMatchSnapshot()
  })
})
