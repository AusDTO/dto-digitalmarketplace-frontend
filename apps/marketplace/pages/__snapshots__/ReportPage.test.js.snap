// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for ReportPageContainer component ReportPageContainer renders without errors 1`] = `
<ReportPageContainer
  data={
    Object {
      "date": null,
      "items": Array [
        Object {
          "heading": "",
          "subitems": Array [],
        },
      ],
      "title": "Marketplace Reports",
    }
  }
  match={[Function]}
/>
`;
